apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts

```diff
@@ -144,42 +173,39 @@ export class DotTemplateStore extends ComponentStore<DotTemplateState> {
                 this.dotGlobalMessageService.loading(
                     this.dotMessageService.get('dot.common.message.saving')
                 );
-                return this.persistTemplate(template);
+                return this.dotTemplateService.update(this.cleanTemplateItem(template));
             }),
-            tap((template: DotTemplate) => this.onSaveTemplate(template)),
-            catchError((err: HttpErrorResponse) => this.onSaveTemplateError(err))
-        );
-    });
-
-    readonly saveTemplateDebounce = this.effect((origin$: Observable<DotTemplateItem>) => {
-        return origin$.pipe(
-            debounceTime(10000),
-            switchMap((template: DotTemplateItem) => {
-                this.dotGlobalMessageService.loading(
-                    this.dotMessageService.get('dot.common.message.saving')
+            tap((template: DotTemplate) => {
+                this.dotGlobalMessageService.success(
+                    this.dotMessageService.get('dot.common.message.saved')
                 );
-                return this.persistTemplate(template);
+                this.updateTemplateState(template);
             }),
-            tap((template: DotTemplate) => this.onSaveTemplate(template)),
             catchError((err: HttpErrorResponse) => this.onSaveTemplateError(err))
         );
     });
 
     readonly saveWorkingTemplate = this.effect((working$: Observable<DotTemplateItem>) => {
         return working$.pipe(
             tap((template: DotTemplateItem) => {
-                if (template.type === 'design') {
+                this.updateWorkingTemplate(template);
+                this.skipSave = false;
+            }),
+            debounceTime(10000),
+            tap((template: DotTemplateItem) => {
+                if (template.type === 'design' && !this.skipSave) {
```

Comment:
![image](https://user-images.githubusercontent.com/37185433/153883862-4587d224-7b77-4c38-87ab-762b14963f46.png)

this task is to add a `Publish` button, so if you make a change on the template and then you click on PUBLISH, we need to skip the auto save functionality with the debounce

Reply:
@alfredo-dotcms I think we just merged these changes but the code we have [here](https://github.com/dotCMS/core-web/blob/f40a2ccf00c4f1308ba6130fd06422e085b09725/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts#L155-L165) is to prevent that behavior.

https://github.com/dotCMS/core-web/blob/f40a2ccf00c4f1308ba6130fd06422e085b09725/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-create-edit/store/dot-template.store.ts#L155-L165